### PR TITLE
Endpoint to get an Api

### PR DESCRIPTION
**Pull request by danifara3**
added Endpoint to get an Api

**Apart from code formatting made the number of changes high

### CODE REVIEW

File: src/api/api.service.ts [hunk 1]
```diff
@@ -32,89 +66,87 @@ export class ApiService {
...
+          'Not found',
+          'The api does not exist',
+          '404',
+        ),
+      );
     }
+    if (api.profileId !== profileId) {
+      throw new NotFoundException(
+        ZaLaResponse.NotFoundRequest(
+          'Not found',
```
Comment (Chimexx) [inline src/api/api.service.ts:98]: There is no need for this. You only have to get a single api by id.

File: src/api/api.controller.ts [hunk 2]
```diff
@@ -7,33 +15,48 @@ import { Api } from '../entities/api.entity';
...
+
+  /**
+   * @Get request that takes
+   * @Param {string} profileId and {string} apiId
+   * @returns a response from the api.service
+   */
+  @Get(':apiId')
+  @ApiOperation({ summary: 'Get an API' })
+  async getAnApi(
+    @Query('profileId') profileId: string,
```
Comment (Chimexx) [inline src/api/api.controller.ts:46]: You don't need profile id to get a single api

**Review by Chimexx — CHANGES_REQUESTED** [review]
I think your task is to get a single api in the db, Getting apis belonging to a particular user has been done by someone else

File: src/api/api.service.ts (hunk 1 above)
Comment (danifara3) [reply src/api/api.service.ts:98]: the reason i added that is to confirm if the profileId has access to the api

**Review by Chimexx — DISMISSED** [review]

**Review by Chimexx — APPROVED** [review]

File: src/api/api.service.ts [hunk 3]
```diff
@@ -8,15 +12,44 @@ import { v4 as uuid } from 'uuid';
...
+    try {
+      const apiExist = await this.apiRepo.findOne({
+        where: { name: createApiDto.name },
+      });
+
+      if (apiExist) {
+        throw new BadRequestException(
+          ZaLaResponse.BadRequest(
+            'Existing values',
+            'An api with with this name already exist... try another name',
```
Comment (johnayinde) [inline src/api/api.service.ts:30]: missing error code

**Review by johnayinde — APPROVED** [review]

**Pull request merged**